README.md

```diff
@@ -9,38 +9,19 @@ on the simplicty of your language's error handling, managing errors and debuggin
 and difficult. For instance, GoLang provides a basic error which contains a simple value - but it
 lacks verbose information about the cause of the error or the state of the stack at error-time.
 
-One solution might be to create a new struct which mimics errors but add extra data. However, this
-solution presents problems for refactoring and future error management, especially if your code base
-is large. Moreover, a custom error object might prevent the easy use of third-party Go libraries,
-most of which use native Go errors.
-
-XErrs uses original GoLang built-in error types and stores all extra data as a string within the
-error itself.
-
-These are the extra extended features which could be extremly useful for any debugging or big
-project:
-
-1. **CauseError** - the original error which happened in the system. This is the original built-in
-   GoLang error which you want to preserve
-2. **MaskError** - a client-facing error. It might be used whenever you want to mask some errors
-   happening on production with a more generic or semantic error
-3. **Stack** - a detailed snapshot of the execution stack at error-time. It might be used to help
-   you to debug errors
-4. **Data** - a mao which contains extra custom data. It could be used for any other data you might
-   want to associate with an error.
-
-## How it works
+XErrs uses original GoLang built-in error types but stores extra data.
 
-To provide added added context to an error, we store an original error object with extra data as a
-serialized object inside a new GoLang error. Using xerrs functions we can extract any relevant data
-from this new error or convert it back to the original error schema.
+These are the extra extended features which could be useful for debugging a big project:
 
-## Bonus Points!
-
-Since the error message is a JSON string it makes very easy to share error objects between systems
-written in different programming languages. For instance, extended error values could be shared by a
-system implemented in Python. Alternatively, your Go code could create an error and then pass it to
-any other systems, regardless of their implementation.
+1. **Cause** - the original error which happened in the system. This is the original error which you
+   want to preserve
+2. **Mask** - a mask error. It might be used whenever you want to mask some errors happening on
+   production with a more generic error. Mask is very useful if you want to prevent original error
+   return back to the client. Once Mask is set there is no way to get to the original error by
```

Comment:
`Once Mask is set there is no way to get to the original error by
   calling Error() function` 
Maybe
```
Calling Error() will always return Mask if set.
```